torch/lib/THD/base/DataChannel.cpp

```diff
@@ -14,17 +14,27 @@
 namespace thd {
 
 DataChannel* DataChannel::newChannel(THDChannelType type) {
-  if (type == THDChannelTCP)
-    return new DataChannelTCP();
+  switch (type) {
+    case THDChannelTCP:
+      return new DataChannelTCP();
+
+    case THDChannelMPI:
 #ifdef WITH_MPI
-  else if (type == THDChannelMPI)
-    return new DataChannelMPI();
+      return new DataChannelMPI();
 #endif // WITH_MPI
+      throw std::runtime_error("the MPI back end is not available; " +
```

Comment:
I'm sorry to inform you that you might need to educate the whole internet about it. I've hardly every seen it spelled as *back end*.